src/app/components/article/ArticlesSidebar.js

```diff
@@ -0,0 +1,225 @@
+    setFlashMessage(errorComponent);
+    setAdding(false);
+  };
+
+  const handleAdd = (nodeType, id) => { // "id" should be an explainer database ID or a fact-check claim description GraphQL ID
+    if (!adding) {
+      setAdding(true);
+      let input = null;
+      let mutation = null;
+      if (nodeType === 'Explainer') {
+        mutation = addExplainerMutation;
+        input = {
+          explainerId: id,
+          projectMediaId: projectMedia.dbid,
+        };
+      } else if (nodeType === 'FactCheck') {
+        mutation = addFactCheckMutation;
+        input = {
+          id,
+          project_media_id: projectMedia.dbid,
+        };
+      }
+      commitMutation(Relay.Store, {
+        mutation,
+        variables: {
+          input,
+        },
+        onCompleted,
+        onError,
+      });
+    }
+  };
+
+  return (
+    <div id="articles-sidebar" className={styles.articlesSidebar}>
+      <div className={styles.articlesSidebarTopBar}>
+        {/* FIXME: Make sure the form can receive the right reference for the current item */}
+        <NewArticleButton team={team} buttonMainProps={{ size: 'small', theme: 'text' }} disabled={projectMedia.type === 'Blank'} />
+      </div>
+      { !hasArticle && (
+        <>
+          <div className={cx('typography-body1', styles.articlesSidebarNoArticle)}>
+            <DescriptionIcon style={{ fontSize: '32px' }} />
+            <div>
+              <FormattedMessage
+                id="articlesSidebar.noArticlesAddedToItem"
+                defaultMessage="No articles are being delivered to Tipline users who send in request with that match this Media."
+                description="Message displayed on articles sidebar when an item has no articles."
+              />
+            </div>
+          </div>
+          <div className="typography-subtitle2">
+            <FormattedMessage
+              id="articlesSidebar.chooseRecentArticle"
+              defaultMessage="Choose a recent article to add to this media:"
+              description="Message displayed on articles sidebar when an item has no articles."
+            />
+          </div>
+          <ArticlesSidebarTeamArticles teamSlug={team.slug} onAdd={handleAdd} />
+        </>
+      )}
+    </div>
+  );
+};
+
+ArticlesSidebarComponent.propTypes = {
+  team: PropTypes.shape({
+    slug: PropTypes.string.isRequired,
+  }).isRequired,
+  projectMedia: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    dbid: PropTypes.number.isRequired,
+    type: PropTypes.string.isRequired,
+    fact_check: PropTypes.object,
+    explainers: PropTypes.exact({
+      edges: PropTypes.arrayOf(PropTypes.object),
+    }),
+  }).isRequired,
+  onUpdate: PropTypes.func.isRequired,
+};
+
+const ArticlesSidebar = ({ teamSlug, projectMediaDbid }) => {
+  const [updateCount, setUpdateCount] = React.useState(0);
+
+  const handleUpdate = () => {
+    setUpdateCount(updateCount + 1);
+  };
```

Comment:
question (non-blocking): why is this necessary?

Reply:
Shouldn't be. But for some reason, for now, the Relay store updates were not enough. I added a `FIXME` to it to see if we can remove it after the rest of the feature is implemented. Once we have the components to list the item's articles, with the proper fragment, I hope we can delete this "hack".

Reply:
Awesome, thanks for clarifying!